dm/master/scheduler/scheduler.go

```diff
@@ -380,7 +380,7 @@ func (s *Scheduler) GetSourceCfgByID(source string) *config.SourceConfig {
 	if !ok {
 		return nil
 	}
-	clone := cfg
+	clone := *cfg
```

Comment:
suddenly found that, because `SourceConfig` contains some reference type member, we can't clone it by this way 😂 

Not checked what changes does the invoker of this function make to `SourceConfig`

https://play.golang.org/p/pwI414VqK0O

Reply:
yes, original code (and seems no effects). 

Not sure we should change it to deepclone or don't clone, in this PR I think left a TODO is OK

Reply:
or let other guy check it 😂